spock-specs/src/test/groovy/org/spockframework/smoke/extension/TempDirExtensionSpec.groovy

```diff
@@ -0,0 +1,158 @@
+package org.spockframework.smoke.extension
+
+import org.spockframework.EmbeddedSpecification
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+/**
+ * @author: dqyuan
+ * @date: 2020/07/12
+ */
+class TempDirExtensionSpec extends EmbeddedSpecification {
+
+  static List<Path> tmpPaths = []
+
+  def setup() {
+    tmpPaths.clear()
+  }
+
+  def assertTmpPathsNotExist() {
+    tmpPaths.each {
+      path -> assert !Files.exists(path)
+    }
+  }
+
+  def "@TempDir creates only one dir for one spec, if the annotated field is 'shared'"() {
```

Comment:
You should rewrite this test as data-driven test without the `runner`.
Then you do not need a `static` field, do not need a nested specification. You can e. g. use a shared field instead and verify that the dir of the previous iteration is the same and still has the content of the first iteration.